fdbserver/KeyValueStoreRocksDB.actor.cpp

```diff
@@ -367,8 +370,13 @@ struct RocksDBKeyValueStore : IKeyValueStore {
 	std::unique_ptr<rocksdb::WriteBatch> writeBatch;
 
 	explicit RocksDBKeyValueStore(const std::string& path, UID id) : path(path), id(id) {
-		writeThread = createGenericThreadPool();
-		readThreads = createGenericThreadPool();
+		if (g_network->isSimulated()) {
+			writeThread = CoroThreadPool::createThreadPool();
```

Comment:
I am confused: does this mean rocks can now be run in a single threaded mode?

Reply:
Not really. The background compaction threads are still present regardless. Also, reads/writes to disk will also block the network thread in a way that would be unacceptable in production but is a necessary evil here. When performing the reads in background threads in simulation, the event loop thinks there is no work to do and advances time faster than 1 sec/sec. By the time the blocking read actually finishes, simulation has advanced time by more than 5 seconds, so every read fails with a `transaction_too_old` error. Doing blocking IO on the main thread solves this issue. There are almost certainly better fixes, but my goal was to get a less invasive change merged first and work on a more realistic version if/when we think that would provide substantially more confidence in the correctness.

Reply:
Maybe add a comment here with the above explanation and TODOs for future improvement.

Reply:
Done